var/spack/repos/builtin/packages/openimageio/package.py

```diff
@@ -30,7 +31,7 @@ class Openimageio(CMakePackage):
     # See https://github.com/spack/spack/pull/22303 for reference
     depends_on(Boost.with_default_variants, type=("build", "link"))
     depends_on("libtiff@4.0:", type=("build", "link"))
-    depends_on("openexr@2.3:", type=("build", "link"))
+    depends_on("openexr@3.2:", type=("build", "link"))
```

Comment:
Should this updated dependency apply to previous versions too? If not, then this needs a `when` clause.

Reply:
Hi, thanks for the feedback! I'm happy to address all of these.
However, the old versions [don't build anymore](https://github.com/spack/spack/issues/42837), so I'm wondering would it make more sense to just remove the old versions from the repo? I don't know anything about compiling libraries so I wouldn't be able to fix the build processes for those.
And what good does it to do complicate the package definition with `when` clauses if those old versions don't build anyway?
If there's a reason to keep the non working older versions around, I'd be happy to just add those when clauses and update the checksums though!
Thank you :)

Reply:
Spack is a bit different than other package managers in that it does not just aim to support the newest version, but also older versions. At some HPC sites, there is a desire to keep these older versions working in the interest of their user base.

When an older version of (in this case) `openimageio` doesn't appear to work anymore with a newer version of `openexr` (which is what it seems like from the linked issue), then this is preferably addressed by adding an upper limit to the version of `openexr` that is allowed for those older versions of `openimageio`.

For example, the following may be a solution here (untested, not looked into the issue in detail):
```py
    depends_on("openexr@2.3:2", type=("build", "link"), when="@1")
    depends_on("openexr@3.2:", type=("build", "link"), when="@2")
```

So, when a build fails, as in the linked issue, we aim to fix the build, irrespective of whether there is a newer version that the user can use. We can also add the newer version, but to some extent that is independent. Similarly, when we do add a newer version, we want to make sure the older versions don't unintentionally stop working.

Finally, there are ways for maintainers to stop supporting older versions, through the `deprecated` attribute for versions. We try to avoid having versions disappear entirely until after there has been a major Spack release with those versions marked as deprecated.

Reply:
Sounds good, I'll try to update the package accordingly, thank you for the detailed instructions!

Reply:
Thanks again for all your help! I really appreciate it, I'm new to `spack` so it's cool to learn how everything works.

I updated all dependencies, including the `url` to work for older versions.
I also handled the TODO regarding the boost dependencies. However, I was unable to get the old packages to build. Even when adding all boost libraries and introducing upper limits for other packages. So now I'm not sure how to move forward with this pull request, I can't get the old versions to build, but I don't have a working reference as they never built successfully for me. Do you have any recommendations on what to do now?

Thank  you :)

Reply:
The old versions can't be built andmore using spack.

Closing this conversation to proceeded with merging.